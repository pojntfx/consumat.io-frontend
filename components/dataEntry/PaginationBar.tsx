import { Dispatch, SetStateAction } from "react";
import {
  ArrowCircleLeftIcon,
  ArrowCircleRightIcon,
} from "@heroicons/react/solid";

type PaginationBarProps = {
  page: number;
  maxPages?: number;
  setPage: Dispatch<SetStateAction<number>>;
};

const PaginationBar = ({ page, maxPages, setPage }: PaginationBarProps) => {
  return (
    <div className="flex mt-6 justify-center items-center">
      <button onClick={() => page > 1 && setPage(page - 1)}>
        <ArrowCircleLeftIcon className="h-8 w-8" />
      </button>
      <p className="font-bold mx-1">
        {page}
        {maxPages && `/${maxPages}`}
      </p>
      <button onClick={() => page < maxPages && setPage(page + 1)}>
        <ArrowCircleRightIcon className="h-8 w-8" />
      </button>
    </div>
  );
};

export default PaginationBar;
