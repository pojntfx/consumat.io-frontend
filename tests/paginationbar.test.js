import React from "react";
import renderer from "react-test-renderer";
import PaginationBar from "../components/dataEntry/PaginationBar";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <PaginationBar
        page={testExamples.examplePage}
        maxPages={testExamples.exampleMaxPages}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
