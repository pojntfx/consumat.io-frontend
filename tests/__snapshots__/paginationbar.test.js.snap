// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex mt-6 justify-center items-center"
>
  <button
    onClick={[Function]}
  >
    <svg
      className="h-8 w-8"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M10 18a8 8 0 100-16 8 8 0 000 16zm.707-10.293a1 1 0 00-1.414-1.414l-3 3a1 1 0 000 1.414l3 3a1 1 0 001.414-1.414L9.414 11H13a1 1 0 100-2H9.414l1.293-1.293z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <p
    className="font-bold mx-1 w-16 text-center"
  >
    1
    /5
  </p>
  <button
    onClick={[Function]}
  >
    <svg
      className="h-8 w-8"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-8.707l-3-3a1 1 0 00-1.414 1.414L10.586 9H7a1 1 0 100 2h3.586l-1.293 1.293a1 1 0 101.414 1.414l3-3a1 1 0 000-1.414z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;
